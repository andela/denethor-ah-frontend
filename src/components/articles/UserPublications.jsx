import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import FeedBottomSmallCard from '../feed/FeedBottomSmallCard';
import { extractImageFromBody } from '../../utils/imageExtractor';
import { toTimeFromNow } from '../../utils/dateTime';


export const UserPublications = (props) => {
  const {
    publishedArticles = [],
  } = props.profile;

  const notFoundContainer = <div>No Article Found</div>;

  const articles = publishedArticles.map(article => {
    article.featuredImage = extractImageFromBody(article.body);
    article.dateCreated =  toTimeFromNow(article.createdAt);
    // article.readTime = '1 minute';
    return article;
  });

  const articlesList = (
    <div className='flex filter-form'>
      {!articles.length && notFoundContainer}
      {!!articles.length && (<h2 className='list-section-header'>Published Articles</h2>)}
      {articles.map((article) => (<FeedBottomSmallCard key={article.id} {...article} />))}
    </div>
  );

  return (
    <div>
      {articlesList}
    </div>
  );
};

UserPublications.propTypes = {
  profile: PropTypes.object,
  history: PropTypes.object,
};

function mapStateToProps ({ profile }) { 
  return {
    profile,
  }
}

export default connect(mapStateToProps)(UserPublications);
