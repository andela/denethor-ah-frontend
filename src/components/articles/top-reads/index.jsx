import TopReads from './TopReads';

export { TopReads };