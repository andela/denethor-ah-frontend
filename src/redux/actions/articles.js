import {
  ADD_ARTICLE,
  REMOVE_ARTICLE,
  GET_ONE_ARTICLE_SUCCESS,
  GET_ARTICLES_SUCCESS,
  RATE_ARTICLE_SUCCESS,
  GET_ARTICLE_AVERAGE_RATING_SUCCESS,
  LIKE_ARTICLE_SUCCESS,
  DISLIKE_ARTICLE_SUCCESS,
  BOOKMARK_ARTICLE_SUCCESS,
} from './types';
import axios from '../../utils/axiosConfig';
import { extractImageFromBody } from '../../utils/imageExtractor';
import { toTimeFromNow } from '../../utils/dateTime';

const { API_ROOT_URL } = process.env;

export const addArticle = article => (dispatch) => {
  dispatch({
    type: ADD_ARTICLE,
    article
  });
};

export const removeArticle = id => (dispatch) => {
  dispatch({
    type: REMOVE_ARTICLE,
    id
  });
};

export const getArticles = (category) => (dispatch) => {
  const query = category && category !== undefined ? `category=${category}` : ''
  return axios
  .get(`${API_ROOT_URL}/articles/?${query}`)
  .then((response) => {
    const articles = response.data.data.map(article => {
      article.featuredImage = extractImageFromBody(article.body);
      article.dateCreated =  toTimeFromNow(article.createdAt);
      return article;
    });
    return dispatch({
      type: GET_ARTICLES_SUCCESS,
      payload: articles
    });
  })
  .catch(error => {
    throw error;
  })
}

export const getOneArticle = id => (dispatch) => {
  return axios.get(`${API_ROOT_URL}/articles/${id}`)
  .then(response => {
    const article = { ...response.data.data };
    article.featuredImage = extractImageFromBody(article.body);
    article.dateCreated =  toTimeFromNow(article.createdAt);
    dispatch({
      type: GET_ONE_ARTICLE_SUCCESS,
      payload: article
    });
  })
  .catch(error => {
    throw error;
  });
};

export const rateArticle = (rating, articleId) => (dispatch) => {
  return axios
  .post(`${API_ROOT_URL}/articles/${articleId}/ratings`, {rating})
  .then(response => {
    dispatch({
      type: RATE_ARTICLE_SUCCESS,
      payload: response.data.article
    })
  })
  .catch((error) => {
    throw error
  })
}

export const getArticleAvgRating = (articleId) => (dispatch) => {
  return axios
  .get(`${API_ROOT_URL}/articles/${articleId}/ratings`)
  .then(response => {
    const { count, rows }= response.data.data
    dispatch({
      type: GET_ARTICLE_AVERAGE_RATING_SUCCESS,
      payload: {
        count, rows, articleId
      }
    })
  })
  .catch((error) => {
    throw error
  })
}

export const likeArticle = (articleId) => (dispatch) => {
  return axios
  .patch(`${API_ROOT_URL}/articles/${articleId}/likes`, {})
  .then(response => {
    dispatch({
      type: LIKE_ARTICLE_SUCCESS,
      payload: response.data.data.message
    });
    return response
  })
  .catch((error) => {
    throw error
  })
}

export const dislikeArticle = (articleId) => (dispatch) => {
  return axios
  .patch(`${API_ROOT_URL}/articles/${articleId}/dislikes`, {})
  .then(response => {
    dispatch({
      type: DISLIKE_ARTICLE_SUCCESS,
      payload: response.data.data.message
    });
    return response
  })
  .catch((error) => {
    throw error
  })
}

export const bookmarkArticle = (articleId) => (dispatch) => {
  return axios
  .post(`${API_ROOT_URL}/articles/${articleId}/bookmark`)
  .then(response => {
    console.log(response)
    dispatch({
      type: BOOKMARK_ARTICLE_SUCCESS,
      payload: response.data
    })
    return response
  })
  .catch((error) => {
    throw error
  })
}
