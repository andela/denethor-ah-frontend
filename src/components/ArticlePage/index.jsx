import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import RoundedImage from '../../components/RoundedImage';
import VerticalMargin from '../../components/VerticalMargin';
import HorizontalMargin from '../../components/HorizontalMargin';
import RatingStars from '../../components/RatingStars';
import RatingStarsBox from '../../components/RatingStarsBox';
import HorizontalLine from '../../components/HorizontalLine';
import TextAreaInput from '../../components/TextAreaInput';
import CommentEntries from '../../components/CommentEntries';
import { getOneArticle, rateArticle, getArticleAvgRating } from '../../redux/actions/articles';
import { toast } from 'react-toastify';
import './style.scss';


export class SingleArticleView extends Component {

  async componentDidMount() {
    const { fetchArticle, getArticleAvgRating, match } = this.props;
    let articleId = match.params.articleId;
    await fetchArticle(articleId);
    getArticleAvgRating(articleId)
  }

  starClickHandle = (item) => {
    let articleId = this.props.match.params.articleId;
    this.props.rateArticle(item, articleId )
    .then(() => toast.success('Thank you for rating this article'))
    .catch(error => {
      if (error.response.data.message === 'You already rated this article'){
        return toast.error('You have already rated this article');
      } else if(error.response.data === 'Unauthorized'){
         toast.error('You need to login to rate an article');
        return  this.props.history.push('/login')
      }
      toast.error('Error while rating article');
    })
  }

  render() {
    const { match: { params: { articleId } }, articles = [] } = this.props;
    const article = articles.find(item => item.id === articleId) || {};
    const { 
      title, 
      featuredImage = '/assets/img/map-typing.jpg', 
      author: { username = '', imageUrl: userAvatarImage = '' } = {}, 
      body,
      averageRating,
      ratingsCount,
      comments = [],
    } = article;

    return (
      <div>
        <div className='banner-section'>
          <div className='banner-image'>
            <img src={featuredImage} alt='banner background' />
          </div>
          <div className='transparent-overlay' />
          <div className='banner-content'>
            <div className='auto-container'>
              <div className='banner-content-title'>{title}</div>
              <div className='avatar-text-group'>
                <div className='user-profile-picture'>
                  <RoundedImage imageSource={userAvatarImage} alt='profile picture'/>
                  <VerticalMargin className='hide-for-medium' size={50} />
                </div>
                <HorizontalMargin className='show-for-medium' size={10} />
                <div className='article-username-rate-group'>
                  <div className='article-header-username'>{username}</div>
                  <div className ='article-rating-container'>
                    <div className='article-header-rating'>
                      <RatingStars rateNumber={(averageRating)} />
                    </div>
                    <div className='article-header-rated-times'>
                      {!ratingsCount ? 'No ratings yet' : `rated by ${ratingsCount} user${ratingsCount > 1 ? 's' : ''}`}
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className='body-section'>
          <div className='auto-container wide-padding'>
            <div className='article-body'>
              <div dangerouslySetInnerHTML={{ __html: body }} />
            </div>
            <HorizontalLine />
            <div className='section-article-info'>  
              <div>Rate this Article <RatingStarsBox starClickHandle= {this.starClickHandle.bind(this)} /></div>  
              <VerticalMargin className='show-for-medium' size={10} />
              <div>
                <div className='section-comment-create'>
                  <div>Leave comments</div>
                  <TextAreaInput className='article-comment-field' text='Your message' />
                  <button></button>
                </div>
                <div className='section-user-comments'>
                  <CommentEntries comments={comments} />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

SingleArticleView.propTypes = {
  articles: PropTypes.array.isRequired,
  match: PropTypes.object.isRequired,
  fetchArticle: PropTypes.func.isRequired,
  rateArticle: PropTypes.func.isRequired,
  getArticleAvgRating: PropTypes.func.isRequired,
  history: PropTypes.object
}

const mapStateToProps = (state) => ({
  articles: state.articles
});

const mapDispatchToProps = (dispatch) => ({
  fetchArticle: (id) => dispatch(getOneArticle(id)),
  rateArticle: (rating, articleId) => dispatch(rateArticle(rating, articleId)),
  getArticleAvgRating: (articleId) => dispatch(getArticleAvgRating(articleId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(SingleArticleView);



