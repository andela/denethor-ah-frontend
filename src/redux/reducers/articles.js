import { ADD_ARTICLE, REMOVE_ARTICLE } from '../actions/types'

const articleReducerDefaultState = [];

export default (state = articleReducerDefaultState, action) => {
  switch (action.type) {
    case ADD_ARTICLE:
      return [...state, action.article];

    case REMOVE_ARTICLE:
      return state.filter(({ id }) => id !== action.id );

    default:
      return state;
  }
};
