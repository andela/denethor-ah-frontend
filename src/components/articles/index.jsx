import ArticlePage from './ArticlePage';
import ArticleCreatePage from './articleCreate/ArticleCreate';

export { ArticlePage, ArticleCreatePage };