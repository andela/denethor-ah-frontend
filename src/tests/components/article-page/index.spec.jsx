import React from 'react';
import { mount } from 'enzyme';
import { SingleArticleView } from '../../../components/articles/ArticlePage';
import articles from '../../mock-data/articles';

describe('Test for the single article component', () => {
	const props = {
		fetchArticle: jest.fn(),
		addComment: jest.fn(),
		match: {
			params: {
				articleId: articles[0].id
			}
		},
		getArticleAvgRating: jest.fn(),
		rateArticle: jest.fn(),
		articles,
		comments: {
			comments: [{
				user: {
					firstname: 'Peace',
					lastname: 'Oyedeji'
				}

			}]
		}
	};

	it('should render single article view page', () => {
		const wrapper = mount(<SingleArticleView {...props} />);
		expect(wrapper.length).toBe(1);
		expect(wrapper.find('.article-body').length).toBe(1);
		expect(wrapper.find('.banner-content-title').length).toBe(1);
		expect(wrapper.find('.user-profile-picture').length).toBe(1);
		expect(wrapper.find('.article-header-username').length).toBe(1);
		expect(wrapper.find('.article-body').length).toBe(1);
	});

	it('should call toast success when starclickhandle is called', () => {
		const rateArticle = jest.fn(() => Promise.resolve({}));
		const props = {
			fetchArticle: jest.fn(),
			addcommment: jest.fn(),
			match: {
				params: {
					articleId: articles[0].id
				}
			},
			getArticleAvgRating: jest.fn(),
			rateArticle,
			articles,
			comments: {
				comments: [{
					user: {
						firstname: 'Peace',
						lastname: 'Oyedeji'
					}
	
				}]
			}
		};
		const wrapper = mount(<SingleArticleView {...props} />);
		wrapper.find('.rating-stars span').at(0).simulate('click', 5);
		expect(wrapper).toMatchSnapshot();
		expect(rateArticle).toHaveBeenCalled();
	});

	it('should call toast error when starclickhandle is called', () => {
		const rateArticle = jest.fn(() => Promise.reject({}));
		const props = {
			fetchArticle: jest.fn(),
			addComment: jest.fn(),
			match: {
				params: {
					articleId: articles[0].id
				}
			},
			getArticleAvgRating: jest.fn(),
			rateArticle,
			articles,
			comments: {
				comments: [{
					user: {
						firstname: 'Peace',
						lastname: 'Oyedeji'
					}
	
				}]
			}
		};
		const wrapper = mount(<SingleArticleView {...props} />);
		wrapper.find('.rating-stars span').at(0).simulate('click', 5);
		expect(wrapper).toMatchSnapshot();
		expect(rateArticle).toHaveBeenCalled();
	});

	it('should like an article', () => {
		const likeArticle = jest.fn(() => Promise.resolve({}));
		const props = {
			fetchArticle: jest.fn(),
			addcommment: jest.fn(),
			match: {
				params: {
					articleId: articles[0].id
				}
			},
			getArticleAvgRating: jest.fn(),
			likeArticle,
			dislikeArticle: jest.fn(),
			bookmarkArticle: jest.fn(),
			rateArticle: jest.fn(),
			articles,
			comments: {
				comments: []
			}
		};
		const wrapper = mount( <SingleArticleView {...props} /> );
		wrapper.find('.like-icon').at(0).simulate('click');
		expect(wrapper).toMatchSnapshot();
		expect(likeArticle).toHaveBeenCalled();
	});

	it('should fail to like an article', () => {
		const likeArticle = jest.fn(() => Promise.reject({}));
		const props = {
			fetchArticle: jest.fn(),
			addcommment: jest.fn(),
			match: {
				params: {
					articleId: articles[0].id
				}
			},
			getArticleAvgRating: jest.fn(),
			likeArticle,
			dislikeArticle: jest.fn(),
			bookmarkArticle: jest.fn(),
			rateArticle: jest.fn(),
			articles,
			comments: {
				comments: []
			}
		};
		const wrapper = mount( <SingleArticleView {...props} /> );
		wrapper.find('.like-icon').at(0).simulate('click');
		expect(wrapper).toMatchSnapshot();
		expect(likeArticle).toHaveBeenCalled();
	});

	it('should dislike an article', () => {
		const dislikeArticle = jest.fn(() => Promise.resolve({}));
		const props = {
			fetchArticle: jest.fn(),
			addcommment: jest.fn(),
			match: {
				params: {
					articleId: articles[0].id
				}
			},
			getArticleAvgRating: jest.fn(),
			likeArticle: jest.fn(),
			dislikeArticle,
			bookmarkArticle: jest.fn(),
			rateArticle: jest.fn(),
			articles,
			comments: {
				comments: []
			}
		};
		const wrapper = mount( <SingleArticleView {...props} /> );
		wrapper.find('.dislike-icon').at(0).simulate('click');
		expect(wrapper).toMatchSnapshot();
		expect(dislikeArticle).toHaveBeenCalled();
	});

	it('should fail to dislike an article', () => {
		const dislikeArticle = jest.fn(() => Promise.reject({}));
		const props = {
			fetchArticle: jest.fn(),
			addcommment: jest.fn(),
			match: {
				params: {
					articleId: articles[0].id
				}
			},
			getArticleAvgRating: jest.fn(),
			likeArticle: jest.fn(),
			dislikeArticle,
			bookmarkArticle: jest.fn(),
			rateArticle: jest.fn(),
			articles,
			comments: {
				comments: []
			}
		};
		const wrapper = mount( <SingleArticleView {...props} /> );
		wrapper.find('.dislike-icon').at(0).simulate('click');
		expect(wrapper).toMatchSnapshot();
		expect(dislikeArticle).toHaveBeenCalled();
	});

});
