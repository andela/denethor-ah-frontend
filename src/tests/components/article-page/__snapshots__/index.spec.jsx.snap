// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the single article component should call toast error when starclickhandle is called 1`] = `
<SingleArticleView
  addComment={[MockFunction]}
  articles={
    Array [
      Object {
        "author": Object {
          "imageUrl": "http://res.cloudinary.com/jsamcloud12/image/upload/v1552059783/profiles/2019-03-08T15:43:03.282Z.jpg",
          "username": "samcotech",
        },
        "body": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. <img src='site.com/image.jpg' /> Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum",
        "categoryId": 1,
        "comments": Array [
          Object {
            "articleId": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
            "commentBody": "Article comme",
            "createdAt": "2019-03-11T15:51:56.685Z",
            "id": "59df4232-0603-44a1-a560-7beffe5569fa",
            "userId": "b590e871-3df2-4cd2-b97f-569c3490b0fa",
          },
        ],
        "createdAt": "2019-03-07T21:05:17.071Z",
        "description": "this is the description",
        "id": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
        "likesCount": 20,
        "readTime": "1 minute",
        "references": null,
        "slug": "this-is-the-title",
        "title": "this is the title",
        "updatedAt": "2019-03-07T21:05:17.071Z",
      },
    ]
  }
  comments={
    Object {
      "comments": Array [],
    }
  }
  fetchArticle={
    [MockFunction] {
      "calls": Array [
        Array [
          "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  getArticleAvgRating={[MockFunction]}
  match={
    Object {
      "params": Object {
        "articleId": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
      },
    }
  }
  rateArticle={
    [MockFunction] {
      "calls": Array [
        Array [
          5,
          "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": Promise {},
        },
      ],
    }
  }
>
  <div>
    <div
      className="banner-section"
    >
      <div
        className="banner-image"
      >
        <img
          alt="banner background"
          src="/assets/img/map-typing.jpg"
        />
      </div>
      <div
        className="transparent-overlay"
      />
      <div
        className="banner-content"
      >
        <div
          className="auto-container"
        >
          <div
            className="banner-content-title"
          >
            this is the title
          </div>
          <div
            className="avatar-text-group"
          >
            <div
              className="user-profile-picture"
            >
              <RoundedImage
                alt="profile picture"
                imageSource="http://res.cloudinary.com/jsamcloud12/image/upload/v1552059783/profiles/2019-03-08T15:43:03.282Z.jpg"
              >
                <div
                  className="rounded-image"
                >
                  <img
                    alt="profile picture"
                    src="http://res.cloudinary.com/jsamcloud12/image/upload/v1552059783/profiles/2019-03-08T15:43:03.282Z.jpg"
                  />
                </div>
              </RoundedImage>
              <VerticalMargin
                className="hide-for-medium"
                size={50}
              >
                <div
                  className="vertical-margin hide-for-medium"
                  style={
                    Object {
                      "height": 50,
                    }
                  }
                />
              </VerticalMargin>
            </div>
            <HorizontalMargin
              className="show-for-medium"
              size={10}
            >
              <div
                className="horizontal-margin show-for-medium"
                style={
                  Object {
                    "marginRight": 10,
                  }
                }
              />
            </HorizontalMargin>
            <div
              className="article-username-rate-group"
            >
              <div
                className="article-header-username"
              >
                samcotech
              </div>
              <div
                className="article-rating-container"
              >
                <div
                  className="article-header-rating"
                >
                  <RatingStars>
                    <span />
                  </RatingStars>
                </div>
                <div
                  className="article-header-rated-times"
                >
                  No ratings yet
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="body-section"
    >
      <div
        className="auto-container wide-padding"
      >
        <div
          className="article-body"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. <img src='site.com/image.jpg' /> Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum",
              }
            }
          />
        </div>
        <HorizontalLine>
          <div
            className="horizontal-line"
          />
        </HorizontalLine>
        <div
          className="section-article-info"
        >
          <div>
            Rate this Article 
            <RatingStarsBox
              starClickHandle={[Function]}
            >
              <span
                className="rating-stars"
              >
                <span
                  key="5"
                  onClick={[Function]}
                >
                  ☆
                </span>
                <span
                  key="4"
                  onClick={[Function]}
                >
                  ☆
                </span>
                <span
                  key="3"
                  onClick={[Function]}
                >
                  ☆
                </span>
                <span
                  key="2"
                  onClick={[Function]}
                >
                  ☆
                </span>
                <span
                  key="1"
                  onClick={[Function]}
                >
                  ☆
                </span>
              </span>
            </RatingStarsBox>
          </div>
          <VerticalMargin
            className="show-for-medium"
            size={10}
          >
            <div
              className="vertical-margin show-for-medium"
              style={
                Object {
                  "height": 10,
                }
              }
            />
          </VerticalMargin>
          <div>
            <div
              className="section-comment-create"
            >
              <div>
                Leave comments
              </div>
              <Comment
                addComment={[MockFunction]}
                articleId="8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a"
              >
                <div
                  className="comment-create-field"
                >
                  <TextAreaInput
                    className="article-comment-field"
                    onChange={[Function]}
                    placeHolder="Your message"
                    placeholder="Your message"
                    value=""
                  >
                    <textarea
                      className="textarea-input article-comment-field"
                      onChange={[Function]}
                      placeholder="Your message"
                      value=""
                    />
                  </TextAreaInput>
                  <Button
                    className="article-comment-field__button"
                    isRequestSent={false}
                    onClick={[Function]}
                    type="submit"
                    value="submit"
                  >
                    <button
                      className="article-comment-field__button"
                      disabled={false}
                      onClick={[Function]}
                      type="submit"
                    >
                      submit
                    </button>
                  </Button>
                </div>
              </Comment>
            </div>
            <div
              className="section-user-comments"
            >
              <CommentEntries
                comments={Array []}
              >
                <div />
              </CommentEntries>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</SingleArticleView>
`;

exports[`Test for the single article component should call toast success when starclickhandle is called 1`] = `
<SingleArticleView
  addcommment={[MockFunction]}
  articles={
    Array [
      Object {
        "author": Object {
          "imageUrl": "http://res.cloudinary.com/jsamcloud12/image/upload/v1552059783/profiles/2019-03-08T15:43:03.282Z.jpg",
          "username": "samcotech",
        },
        "body": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. <img src='site.com/image.jpg' /> Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum",
        "categoryId": 1,
        "comments": Array [
          Object {
            "articleId": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
            "commentBody": "Article comme",
            "createdAt": "2019-03-11T15:51:56.685Z",
            "id": "59df4232-0603-44a1-a560-7beffe5569fa",
            "userId": "b590e871-3df2-4cd2-b97f-569c3490b0fa",
          },
        ],
        "createdAt": "2019-03-07T21:05:17.071Z",
        "description": "this is the description",
        "id": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
        "likesCount": 20,
        "readTime": "1 minute",
        "references": null,
        "slug": "this-is-the-title",
        "title": "this is the title",
        "updatedAt": "2019-03-07T21:05:17.071Z",
      },
    ]
  }
  comments={
    Object {
      "comments": Array [],
    }
  }
  fetchArticle={
    [MockFunction] {
      "calls": Array [
        Array [
          "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  getArticleAvgRating={[MockFunction]}
  match={
    Object {
      "params": Object {
        "articleId": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
      },
    }
  }
  rateArticle={
    [MockFunction] {
      "calls": Array [
        Array [
          5,
          "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": Promise {},
        },
      ],
    }
  }
>
  <div>
    <div
      className="banner-section"
    >
      <div
        className="banner-image"
      >
        <img
          alt="banner background"
          src="/assets/img/map-typing.jpg"
        />
      </div>
      <div
        className="transparent-overlay"
      />
      <div
        className="banner-content"
      >
        <div
          className="auto-container"
        >
          <div
            className="banner-content-title"
          >
            this is the title
          </div>
          <div
            className="avatar-text-group"
          >
            <div
              className="user-profile-picture"
            >
              <RoundedImage
                alt="profile picture"
                imageSource="http://res.cloudinary.com/jsamcloud12/image/upload/v1552059783/profiles/2019-03-08T15:43:03.282Z.jpg"
              >
                <div
                  className="rounded-image"
                >
                  <img
                    alt="profile picture"
                    src="http://res.cloudinary.com/jsamcloud12/image/upload/v1552059783/profiles/2019-03-08T15:43:03.282Z.jpg"
                  />
                </div>
              </RoundedImage>
              <VerticalMargin
                className="hide-for-medium"
                size={50}
              >
                <div
                  className="vertical-margin hide-for-medium"
                  style={
                    Object {
                      "height": 50,
                    }
                  }
                />
              </VerticalMargin>
            </div>
            <HorizontalMargin
              className="show-for-medium"
              size={10}
            >
              <div
                className="horizontal-margin show-for-medium"
                style={
                  Object {
                    "marginRight": 10,
                  }
                }
              />
            </HorizontalMargin>
            <div
              className="article-username-rate-group"
            >
              <div
                className="article-header-username"
              >
                samcotech
              </div>
              <div
                className="article-rating-container"
              >
                <div
                  className="article-header-rating"
                >
                  <RatingStars>
                    <span />
                  </RatingStars>
                </div>
                <div
                  className="article-header-rated-times"
                >
                  No ratings yet
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="body-section"
    >
      <div
        className="auto-container wide-padding"
      >
        <div
          className="article-body"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. <img src='site.com/image.jpg' /> Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum",
              }
            }
          />
        </div>
        <HorizontalLine>
          <div
            className="horizontal-line"
          />
        </HorizontalLine>
        <div
          className="section-article-info"
        >
          <div>
            Rate this Article 
            <RatingStarsBox
              starClickHandle={[Function]}
            >
              <span
                className="rating-stars"
              >
                <span
                  key="5"
                  onClick={[Function]}
                >
                  ☆
                </span>
                <span
                  key="4"
                  onClick={[Function]}
                >
                  ☆
                </span>
                <span
                  key="3"
                  onClick={[Function]}
                >
                  ☆
                </span>
                <span
                  key="2"
                  onClick={[Function]}
                >
                  ☆
                </span>
                <span
                  key="1"
                  onClick={[Function]}
                >
                  ☆
                </span>
              </span>
            </RatingStarsBox>
          </div>
          <VerticalMargin
            className="show-for-medium"
            size={10}
          >
            <div
              className="vertical-margin show-for-medium"
              style={
                Object {
                  "height": 10,
                }
              }
            />
          </VerticalMargin>
          <div>
            <div
              className="section-comment-create"
            >
              <div>
                Leave comments
              </div>
              <Comment
                articleId="8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a"
              >
                <div
                  className="comment-create-field"
                >
                  <TextAreaInput
                    className="article-comment-field"
                    onChange={[Function]}
                    placeHolder="Your message"
                    placeholder="Your message"
                    value=""
                  >
                    <textarea
                      className="textarea-input article-comment-field"
                      onChange={[Function]}
                      placeholder="Your message"
                      value=""
                    />
                  </TextAreaInput>
                  <Button
                    className="article-comment-field__button"
                    isRequestSent={false}
                    onClick={[Function]}
                    type="submit"
                    value="submit"
                  >
                    <button
                      className="article-comment-field__button"
                      disabled={false}
                      onClick={[Function]}
                      type="submit"
                    >
                      submit
                    </button>
                  </Button>
                </div>
              </Comment>
            </div>
            <div
              className="section-user-comments"
            >
              <CommentEntries
                comments={Array []}
              >
                <div />
              </CommentEntries>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</SingleArticleView>
`;
